sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -334,6 +389,25 @@ class SQLViewSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  test("correctly handle ALTER VIEW on a referenced view") {
+    withTable("jt2") {
+      val df = (1 until 10).map(i => (i, i + 1)).toDF("i", "j")
+      df.write.format("json").saveAsTable("jt2")
+
+      withView("view1", "view2") {
+        sql("CREATE VIEW view1(x, y) AS SELECT * FROM jt")
+
+        // Create a nested view.
+        sql("CREATE VIEW view2(id, id1) AS SELECT * FROM view1")
+        checkAnswer(sql("SELECT * FROM view2 ORDER BY id"), (1 to 9).map(i => Row(i, i)))
+
+        // Alter the referenced view.
+        sql("ALTER VIEW view1 AS SELECT i AS x, j AS y FROM jt2")
```

Comment:
nit: `ALTER VIEW view1 AS SELECT id AS x, id1 + 1 As y FROM jt`, then we don't need to create `jt2`